src/item_menu_icons.c

```diff
@@ -544,7 +544,20 @@ void AddBagItemIconSprite(u16 itemId, u8 id)
 
 void RemoveBagItemIconSprite(u8 id)
 {
-    RemoveBagSprite(id + ITEMMENUSPRITE_ITEM);
+    #ifdef BUGFIX
+        u8 *spriteId = &gBagMenu->spriteIds[ITEMMENUSPRITE_ITEM];
+
+    if (spriteId[id ^ 1] != SPRITE_NONE)
+        gSprites[spriteId[id ^ 1]].invisible = TRUE;
+
+    if (spriteId[id] != SPRITE_NONE)
+    {
+        DestroySpriteAndFreeResources(&gSprites[spriteId[id]]);
+        spriteId[id] = SPRITE_NONE;
+    }
+    #else
+        RemoveBagSprite(id + ITEMMENUSPRITE_ITEM);
+    #endif
```

Comment:
```suggestion
#ifdef BUGFIX
    u8 *spriteId = &gBagMenu->spriteIds[ITEMMENUSPRITE_ITEM];

if (spriteId[id ^ 1] != SPRITE_NONE)
    gSprites[spriteId[id ^ 1]].invisible = TRUE;

if (spriteId[id] != SPRITE_NONE)
{
    DestroySpriteAndFreeResources(&gSprites[spriteId[id]]);
    spriteId[id] = SPRITE_NONE;
}
#else
    RemoveBagSprite(id + ITEMMENUSPRITE_ITEM);
#endif
```

Reply:
I was gonna talk about the syntax too, but then I noticed it's already inconsistent.

Check `src/daycare.c` and `src/battle_script_commands.c` for example.